### PR TITLE
CLI: Improve yarn 1 vs Berry detection and log package manager during init

### PR DESCRIPTION
**Pull request by Sidnioulz**
- [x] Fix yarn 1 version detection regex (`/^1\./`)
- [x] Add Berry detection via `packageManager` field in `package.json`
- [x] Add Berry detection via `.yarnrc.yml` presence
- [x] Log package manager during `sb init`
- [x] Add unit tests for new detection logic
- [x] Apply comment suggestion from @Sidnioulz (improved inline comment wording)
- [x] Compute `getProjectRoot()` once in `getYarnVersion()` and pass result to helper functions to avoid repeated filesystem traversals

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved Yarn detection: now considers package.json's packageManager and presence of Yarn Berry config in ancestor folders in addition to yarn --version, yielding more accurate Yarn 1/2/Berry identification.

* **New Features**
  * The selected package manager is now logged during initialization with a friendly label.

* **Tests**
  * Added extensive tests covering upward package.json walking, Yarn Berry config scenarios, and various Yarn version/reporting edge cases.

### CODE REVIEW

File: code/core/src/common/js-package-manager/JsPackageManagerFactory.ts [hunk 1]
```diff
@@ -259,8 +306,20 @@ function getYarnVersion(cwd?: string): 1 | 2 | undefined {
         Object.entries(process.env).filter(([, value]) => value !== undefined)
       ) as Record<string, string>,
     });
-    return /^1\.+/.test(yarnVersion.trim()) ? 1 : 2;
+
+    if (/^1\./.test(yarnVersion.trim())) {
+      // yarn --version reports 1.x, but .yarnrc.yml means it's actually Berry
```
Comment (Sidnioulz) [inline code/core/src/common/js-package-manager/JsPackageManagerFactory.ts]:
```suggestion
      // yarn --version reports 1.x, but .yarnrc.yml means it's actually Berry; happens if the user's global Yarn is used because they forgot to enable corepack
```

File: code/core/src/common/js-package-manager/JsPackageManagerFactory.ts [hunk 2]
```diff
@@ -249,7 +250,53 @@ function hasPNPM(cwd?: string) {
...
+      const match = packageManager.match(/^yarn@(\d+)\./);
+      if (match) {
+        return match[1] === '1' ? 1 : 2;
+      }
+    }
+  } catch {
+    // Ignore parse errors
+  }
+
+  return undefined;
```
[the remark this answers is not in the document]
Comment (yannbf) [reply code/core/src/common/js-package-manager/JsPackageManagerFactory.ts]: This is a really good point @Sidnioulz

[the remark this answers is not in the document]
Comment (Sidnioulz) [reply code/core/src/common/js-package-manager/JsPackageManagerFactory.ts]: @copilot 
Verify each finding against the current code and only fix it if needed.

In `@code/core/src/common/js-package-manager/JsPackageManagerFactory.ts` around
lines 257 - 277, getYarnVersionFromPackageJson currently stops at the first
package.json found (using find.up) even if that file lacks a packageManager
field; change it to walk upward from the project root from getProjectRoot() (fallback to cwd)
and check every package.json until one declares packageManager. Concretely, in
getYarnVersionFromPackageJson: start at getProjectRoot() (or cwd), loop up directories until reaching the project
root, at each directory check for package.json (e.g., using find.up or
fs.existsSync/readFileSync), parse and inspect content.packageManager, and
return 1 or 2 when a matching "yarn@X." is found; only return undefined after
the loop completes. Keep the same error handling (try/catch around JSON.parse)
and continue up on missing or unparsable package.json files.

Add unit tests showcasing both that project root is used, that cwd is the fallback, and the priority order of reads when multiple package.json are defined

File: code/core/src/common/js-package-manager/JsPackageManagerFactory.ts [hunk 3]
```diff
@@ -249,7 +250,53 @@ function hasPNPM(cwd?: string) {
...
 function getYarnVersion(cwd?: string): 1 | 2 | undefined {
+  // 1. Check packageManager field in closest package.json (highest priority)
+  const versionFromPackageJson = getYarnVersionFromPackageJson(cwd);
+  if (versionFromPackageJson !== undefined) {
+    return versionFromPackageJson;
+  }
+
+  // 2. Check for .yarnrc.yml (Berry-only config file)
+  const hasBerryConfig = hasYarnBerryConfig(cwd);
+
```
[the remark this answers is not in the document]
Comment (Sidnioulz) [reply code/core/src/common/js-package-manager/JsPackageManagerFactory.ts]: @copilot fix this code smell

**Comment by Sidnioulz** [comment]
@yannbf should be good to re-review now, I've addressed the bunny feedback

**Review by yannbf — APPROVED** [review]
LGTM! As long as sandbox generation works well I'm good with it!
